### PR TITLE
Enhancement: Update session content and session structure for activities session 0x2

### PR DESCRIPTION
**Pull request by gabrielmocanu**
Enhancement: Update session content and session structure for activities for session 0x2

### CODE REVIEW

**Comment by gabrielmocanu** [comment]
@carrrina  can you look on this PR?

File: cookies-session-management-access-control/README.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+By using the command you can find a list of sites that expose their _.env_ file on the internet.
+Developers may accidentally include the _.env_ file in the public directory of the website, which can cause great harm if cyber criminals find it.
+
+If you click into any of the exposed .env files, you will notice unencrypted usernames, passwords and IPs are directly exposed in the search results.
+**Prevention:** _.env_ files should **not** be in a publicly accessible folder.
+
+## Wayback Machine
+
+The Wayback Machine is a digital archive of the entire internet.
```
Comment (carrrina) [inline cookies-session-management-access-control/README.md:273]:
```suggestion
The [Wayback Machine](https://archive.org/web/) is a digital archive of the entire internet.
```

**Comment by carrrina** [comment]
The CORS.jpg image does not appear to be used

**Comment by carrrina** [comment]
Same note: it would be nice to have a README solution for each challenge

**Comment by carrrina** [comment]
All the `solution` scripts should be named `solution.sh` as per [here](https://github.com/security-summer-school/template-internal)

File: cookies-session-management-access-control/README.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+3. When the client changes the page, it sends all the cookies in the request, along with the session ID from step 1.
+4. The server reads the session ID from the cookie.
+5. The server matches the session ID with the entries of a local list (in-memory, text file etc.).
+6. If the server finds a match, it reads the stored variables. For PHP, these variables will become available in the superglobal variable `$_SESSION`.
+7. If the server doesn’t find a match, it will create a new session and repeat steps 1-6.
+
+![Session lifecycle](./assets/session.jpg)
+
+Example of a session in PHP:  
+`<?php`
```
[the remark this answers is not in the document]
Comment (gabrielmocanu) [reply cookies-session-management-access-control/README.md:86]: Done.

File: cookies-session-management-access-control/README.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+Example of a session in PHP:  
+`<?php`  
+`session_start(); // Start the session`  
+`$_SESSION['username'] = "John Doe";`  
+`$_SESSION['is_admin'] = true;`  
+`echo "Hello " . $_SESSION['username'];`  
+`?>`
+
+Example of a session in Python:  
+`s = requests.Session()`
```
[the remark this answers is not in the document]
Comment (gabrielmocanu) [reply cookies-session-management-access-control/README.md:94]: Done.

File: cookies-session-management-access-control/README.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+
+- The application should validate the user input before processing it.
+  Ideally, the validation should compare against a whitelist of permitted values.
+  If that isn't possible for the required functionality, then the validation should verify that the input contains only permitted content, such as purely alphanumeric characters.
+- After validating the supplied input, the application should append the input to the base directory and use a platform filesystem API to canonicalize the path.
+  It should verify that the canonicalized path starts with the expected base directory.
+
+Below is an example of some simple Java code to validate the canonical path of a file based on user input:
+
+`File file = new File(BASE_DIRECTORY, userInput);`
```
[the remark this answers is not in the document]
Comment (gabrielmocanu) [reply cookies-session-management-access-control/README.md:184]: Done.

File: cookies-session-management-access-control/README.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+One fairly common hacker trick is to view a site’s robots.txt file first to get an idea of how files and directories are mapped on the server.
+In fact a quick Google query can reveal lots of sites that have had their robots.txt files crawled.
+This, of course, is a misconfiguration, because the robots.txt file is meant to stay behind the scenes.
+
+## Sitemap.xml
+
+The sitemap.xml is a simple XML page which could be available on some websites and provide a “roadmap” for Google to the important pages that need to be crawled.
+It’s a SEO (Search Engine Optimization) tool to help with the visibility of your website on the internet, but it could also be useful for a hacker, serving basically the same purpose: to give him a roadmap to every page.
+
+## Examples of Google Dorking
```
[the remark this answers is not in the document]
Comment (gabrielmocanu) [reply cookies-session-management-access-control/README.md:248]: Good ideas. Did it!

File: cookies-session-management-access-control/activities/beep-beep-boop/sol/solution [hunk 6]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+PORT=8088
+
+if [[ $1 == "local" ]]
+then
+    remote='http://127.0.0.1:'$PORT
```
[the remark this answers is not in the document]
Comment (gabrielmocanu) [reply cookies-session-management-access-control/activities/beep-beep-boop/sol/solution]: I did this change for all solution scripts.

**Comment by gabrielmocanu** [comment]
@carrrina 

I did all the changes that you suggested.
Please have a look again. Thanks!

File: cookies-session-management-access-control/README.md [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+
+### Google Hacking tool from [Pentest-Tools](https://pentest-tools.com/information-gathering/google-hacking)
```
Comment (carrrina) [inline cookies-session-management-access-control/README.md:256]: I suggest putting these two in a bulleted list for more legibility

**Comment by carrrina** [comment]
You can delete the old session folder and update the repo README to point to this new one in this same PR.

**Review by carrrina — CHANGES_REQUESTED** [review]

File: cookies-session-management-access-control/README.md (hunk 7 above)
Comment (gabrielmocanu) [reply cookies-session-management-access-control/README.md]: Done.

**Comment by gabrielmocanu** [comment]
@carrrina 

Can you take a look again, please?

**Review by carrrina — APPROVED** [review]

**Pull request merged**